user_list: Distinguish between full and new members.

Jagansivam28 opened the pull request:
Previously, in the admin user list,user roles were listed as just "member"
so it's difficult to distinguish between full members and new members.

Now, the user's roles were listed as  "Full Member" or "New Member"
in admin user list rather than just "Member".

Also in this commit changes were made to distinguish between
full member and new member in other places like
`user profile` , `user info popover` etc.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->
fixes #16040

**Testing Plan:** <!-- How have you tested? -->
Tested using automated node testes and manually tested in the browser

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented on static/js/people.js:
```diff
@@ -59,6 +59,14 @@ exports.get_by_user_id = function (user_id, ignore_missing) {
+    const person_date_joined = new Date(person.date_joined);
+    const days = (current_datetime - person_date_joined) / 1000 / 86400;
+    return days > page_params.realm_waiting_period_threshold;
+};
```
Please do this refactoring in its own preparatory commit; it'll be a lot more readable that way.

Jagansivam28 replied:
ok sure Tim

timabbott commented on static/templates/admin_user_list.hbs:
```diff
@@ -17,8 +17,10 @@
+            {{else if is_full_member }}
+            {{t "Full member" }}
+            {{else }}
+            {{t "New member" }}
```
We should make sure this page has a link to `/help/roles-and-permissions`, maybe just next to the "Users" label, with our usual settings /help link system.

(Can you do this in a prep commit?)

Jagansivam28 replied:
ok sure

Jagansivam28 replied:
also we need to distingush between the full and new member  in the docs `/help/roles-and-permissions` right?

Jagansivam28 committed:
user: Create a new function to check full member or new member.

In this commit created a new function to check for a full member
or new member that will use to distinguish between the full member
and new member in future.
Checking  full member or a new member is accomplished by comparing
the  `realm_waiting_period_threshold` days with number of days after
the joined date.

Jagansivam28 committed:
user_list: Distinguish between full and new members.

Previously, in the admin user list,user roles were listed as just "member"
so it's difficult to distinguish between full members and new members.

Now, the user's roles were listed as  "Full Member" or "New Member"
in admin user list rather than just "Member".

Also in this commit changes were made to distinguish between
full member and new member in other places like
`user profile` , `user info popover` etc.

Jagansivam28 committed:
user_list: Add link to docs `/help/roles-and-permissions` in User list.

Since we distinguish between full member and new members in user list
it is necessary to add a link for user `role and permissions` docs
in user list settings.

fixes #16040

Jagansivam28 commented:
@timabbott ready for another review
